source/core/utilities.cc

```diff
@@ -94,7 +94,113 @@ make_table_tensors_scalars(
   return table;
 }
 
+void
+fill_table_from_file(TableHandler &    table,
+                     std::string       file,
+                     const std::string delimiter)
+{
+  table.clear();
+  std::ifstream myfile(file);
+  // open the file
+  if (myfile.is_open())
+    {
+      std::string              line;
+      std::vector<std::string> vector_of_column_names;
+      std::vector<double>      line_of_data;
+      unsigned int             i = 0;
+
+      while (std::getline(myfile, line))
+        {
+          // read the line and clean the resulting vector
+          std::vector<std::string> list_of_words_base =
+            Utilities::split_string_list(line, delimiter);
+          std::vector<std::string> list_of_words_clean;
+          for (unsigned int i = 0; i < list_of_words_base.size(); ++i)
+            {
+              if (list_of_words_base[i] != "")
+                {
+                  list_of_words_clean.push_back(list_of_words_base[i]);
+                }
+            }
+          //  If it's the first line, we only initialize the variable names.
+          if (i != 0)
+            {
+              line_of_data = Utilities::string_to_double(list_of_words_clean);
+              for (unsigned int i = 0; i < line_of_data.size(); ++i)
+                {
+                  table.add_value(vector_of_column_names[i], line_of_data[i]);
+                }
+            }
+          else
+            {
+              // the line contains words we assume these are the column
+              vector_of_column_names = list_of_words_clean;
+            }
+          ++i;
+        }
+      myfile.close();
+    }
+  else
+    std::cout << "Unable to open file";
+}
+
+void
+fill_vectors_from_file(std::map<std::string, std::vector<double>> &map,
+                       std::string                                 file,
+                       const std::string                           delimiter)
+{
+  // fill a pair, first being a vector of vector name and the second being the
+  // vector of vector associated with the vector name.
+
+
+  std::ifstream myfile(file);
+  // open the file.
+  if (myfile.is_open())
+    {
+      std::string              line;
+      std::vector<std::string> column_names;
+      std::vector<double>      line_of_data;
+      unsigned int             i = 0;
```

Comment:
Same comment as above, use another variable line to make sure both can be distinguished.